### ISSUE TITLE
More reasons to go to the spaceport, and more to do there

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#1552]
Right now the only reason to go into the spaceport is the chance to get a mission. If you don't, then there's just a bit of text--nothing else to do.

In the old Escape Velocity games, the spaceport bar also gave you the opportunity to hire escorts, gamble, or read the news. Stuff like this would be great. Additional ideas:
* Buy some old spacers a drink to hear local gossip
* Get in a fight with pirates over something that causes you to gain some money, but your ship immediately takes off along with some weak pirate interceptors
* Look for opportunities to buy illegal outfits
* Look for opportunities to transport illegal goods

**Comment by Amazinite** [comment]
A similar idea I had a while back: random events that can occur throughout human space that the player can only know about if they run into them, or if they decide to visit the spaceport to check out the news board.

Such events could be a sudden increase in piracy in a certain part of space (with bounties over the heads of a few pirates), trade changes between two parts of space that affect commodity prices (so pretty much commodity fluctuations to one or two commodities that don't require the player to be trading a lot to occur), and to go along with your Firefly idea of increasing how dangerous the universe is, have an event where a star may start having a massive spike in coronal activity that irradiates the system for a few days, lowering the fleet frequencies in the system and causing light damage when in the system.

**Comment by jafdy** [comment]
Every thing you want can be accomplished with more missions that are taken in the space port.

**Comment by Pointedstick** [comment]
Yes, but only very awkwardly, in much the same manner that you could (and people did) add asteroid mining by making a government whose ships looked like asteroids that dropped cargo when destroyed that looked like various minerals. The native implementation is much smoother and better.

**Comment by jafdy** [comment]
Certainly points 2 and 4 can. eg
Drug Running missions
`mission "fight pirates"
on offer
   conversation
      Some pirates are gambling in the bar. Would you like to join in?
      choice
         yes
         no
            decline
      You win some money but the pirates now want to kill you.
         accept
on accept
   payment 1000
fleet "small pirate"`

For point 1 it would be an auto decline mission with a negative payment and some text.

Fair enough on point 3.

Hiring escorts is mostly a matter of how to do it. I set something up but the negative payments don't work properly in the job board.  

With gambling (not in above mission) and the things @Amazinite mentioned a new mechanic would probably be best.

**Comment by Amazinite** [comment]
The events are, in a way, doable (although, the addition of new mechanics would certainly make it easier as to not have to make an event for every single system and every single event type); it's the news board idea that would require new mechanics to not make it so awkward.

**Comment by Amazinite** [comment]
@Pointedstick Do you still think that the spaceport could use more spice? We now have the spaceport news system and we've added a variety of [smaller missions/conversations](https://github.com/endless-sky/endless-sky/blob/master/data/human/culture%20conversations.txt) meant to give the spaceport more life.

**Comment by Zitchas** [comment]
The Spaceport Bar is better in some regards - namely stuff happens there more often.

Very little in the original request/issue has changed: There's still nothing that we can *do* in the Spaceport. It's still just a quick click, see if anything triggers, and move on. Beyond that, given the way it is presented, the spaceport doesn't feel like somewhere we *go*, and definitely not somewhere that we actually *do* things; it's just an alternate description for the planet we are sitting on. 

Back in EV/O/N, it was actually a place we visited (as the pop-up dialog box made it clear, since it opened up new buttons to click and we had to actually *leave* it in order to co to the outfitter or trading or shipyard.) It was also a useful place to visit because one could reliably get information on where price spikes/drops were happening, for instance. 

The hiring escorts part I very much thematically liked better in the predecessor games, since we were clearly hiring an escort, and each ship had a description suitable for it. For one, it had a lot more character than our current "buy a ship and just choose to not fly it" system. For two, hiring a ship gave players limited access to ships that they might otherwise can't afford. For example, as a fairly early game player they could hire a bulk freighter to get the extra cargo space to do some commodity runs. The hiring rate is probably double or triple what the regular crew wages would be, and the player isn't allowed to take control of it (so, very much like a mission provided escort), but they otherwise get the benefit of having it in their fleet. But, they don't have to actually *buy* the ship. This shifts the focus away from buying/capturing a fleet to handle situational needs for extra ships, since one can just hire a ship for a few days or however long as needed.

The gambling was just a money trap, as gambling tends to be. Although I vaguelly recall there was a pattern so if one did it correctly one could usually make money at it. It was an interesting thing to do once in a while just for variety, but it didn't really add much to the game. What would make it much more useful (and worth considering for addition) is if gambling counted as taking off, landing, and visiting the spaceport bar again. Effectively a button to pass time and reroll the planet (jobs get re-generated, odds for missions get rerolled, etc). Something more interesting than gambling might be better, though.

tldr:
No, this is not resolved. Higher odds that something happens to the player isn't the same thing as consistently having something for the player to do.

**Comment by bene-dictator** [comment]
Regarding the news system... I still think Pointedstick's original issue stands (#3389, #3366). I don't think it does very much at all to make the spaceport a distinct place; it's mostly some detached text from some random person you've met in the spaceport, or sometimes something you've read in the spaceport (Graffiti? Poems? Health and Safety signs?)

What I think would be better is a scrolling news ticker. MZ vetoed that in [#3389 (comment)](https://github.com/endless-sky/endless-sky/issues/3389#issuecomment-352062699) even though some people said it would be better than the news system he eventually implemented (not the first time!) I always liked how Naev did its news in their spaceport bar, and I'd enjoy writing those dynamic news stories myself. When it comes to _news_ stories (as oppossed to random text from random people like above), the current news system is not very "staying" since you can just flip through them by clicking "spaceport", and there's not too much room in the box to display text. The same way that most people ignore NPC hails, why should people pay attention to text that is pretty obviously randomly generated, given the fact that the game just makes them on the fly with a click of a button? Whereas a news ticker shows news from the past week or two, featuring lasting stories of both regional and galactic scale that the player can read back upon, in a fashion that's more visible and more reactive than your logbook. A few examples:
* New players could be introduced to the rising tension in the Southern Rim in a fashion less in-your-face than a conversation (some examples of what could be covered can be found in #9627); they can look back at these news stories if necessary, or be reminded by continuous news coverage (e.g. developments after the bombings like failed negotiations or declining trade due to FW-Republic embargoes)
* Remind players that they can join factions – Navy looking for new officers, Syndicate looking for new captains, FW looking for militia members, etc.
* Current depictions of news stories in-game could be represented with this news system; for example, the Wolf Pack's hijacking of the news outlets in the FW, the media coverage of the Hai and then their retraction, and then coverage of the Hai-human treaty
* Players can be reminded of their previous deeds. For example, there oculd be coverage of the outrage against the player smashing an asteroid into Rand, then their summons to Earth, and then the improvements in the conditions of Rand and how it turns into a tourist destination
* And of course, little tid-bits of lore can be sprinkled around, each relating to their planet/region. e.g. "Feral goat population found outside Big Sky"; "Mars denied increased terraforming subsidies in Parliament for 174th time"; "New type of ore speculation increases prospects on Gemstone"; "Syndicate denounces special report claiming it has ties to slave cartels"; "Climate scientists begin attempt to diminish dust storms on Crossroads"; "Increased pirate raids from southern anarchist worlds 'not a concern', says councillor" etc.

Of course, this system may be in conflict with the current "news" system, and besides, apart from the Hai, creating news stories for aliens would be a challenge.

Naev's system that shows headlines and a short blurb of what's happened:
<img width="461" alt="Screenshot 2024-02-03 at 7 01 59 PM" src="https://github.com/endless-sky/endless-sky/assets/115441627/5a5fdc78-8748-4aae-b914-e767b0870783">

Something I made myself a few months ago by replacing the credits in the main menu - probably too long but just for demonstration:

https://github.com/endless-sky/endless-sky/assets/115441627/01f7a5e0-1063-432f-8fc5-f9ddf08a5b17

I think the culture convos tend to be a good thing, albeit everyone who encounters them seem to either be wondering if they missed a mission or are just looking for the Mug™.

**Comment by Pointedstick** [comment]
I agree with what @Zitchas and @bene-dictator have said. I don't really consider this solved now.

**Comment by Amazinite** [comment]
Shower thought I just had: what if clicking on the spaceport button changed the options you had for the other buttons?

Landing:
<img width="892" height="757" alt="Image" src="https://github.com/user-attachments/assets/7d62e44f-77b6-4173-b304-7c69ad718f82" />

"In" the spaceport:
<img width="919" height="765" alt="Image" src="https://github.com/user-attachments/assets/c29cd996-5698-45da-bfab-6587aa860c90" />

The options that show up could be "attractions" that are a part of the spaceport. Different planets could have different attractions. So not every spaceport would have the option to go gambling, for example, or might have a different number of attractions. Attractions should be able to do multiple things:
1. Change the landscape image and the text in the description box.
2. Trigger missions, and we could tie particular missions to particular attractions. Gambling could be a repeatable mission for certain planets, for example.
3. Change the "news" blurb (randomize from the usual list or maybe even have its own list of options to choose from).

I imagine it would be difficult to flesh out all the planets we have with interesting attractions, but if this doesn't give a good reason to visit the spaceport, I don't know what else would.

**Comment by Pointedstick** [comment]
Yes, that's a lot like what I originally had in mind!

**Comment by nickshanks** [comment]
> Every thing you want can be accomplished with more missions that are taken in the space port.

More bar missions won't make it look pretty.
